### DISCUSSION TITLE
Basic arithmetic lemma can't be proved

### DISCUSSION THREAD

**Discussion by SeanCoek** [dafny-lang/dafny#5964] [Q&A]
I am starting to learn proof in Dafny. But I don't know why Dafny can't prove this, cause it seems quite straightforward.

lemma arithTest(a:int, b:int)
requires -1 <= a <= b
ensures exists i:int :: b - a == i
{
}

**Comment by racko** [comment]
See https://github.com/dafny-lang/dafny/discussions/5563

```dfy
predicate P(a: int, b: int, i: int)
{
    b - a == i
}

lemma arithTest(a: int, b: int)
    requires -1 <= a <= b
    ensures exists i: int :: P(a, b, i)
{
    assert P(a, b, b - a);
}
```

More qualified people should answer _why_ we have to help Dafny like this :slightly_smiling_face:
